src/clj/fluree/db/query/exec/select.cljc

```diff
@@ -52,16 +53,19 @@
 
 (defprotocol ValueSelector
   (implicit-grouping? [this]
-   "Returns true if this selector should have its values grouped together.")
+    "Returns true if this selector should have its values grouped together.")
   (format-value [fmt db iri-cache context compact fuel-tracker error-ch solution]
-   "Formats a where search solution (map of pattern matches) by extracting and displaying relevant pattern matches."))
+    "Async format a search solution (map of pattern matches) by extracting relevant match.")
+  (solution-value [fmt error-ch solution]
+    "Formats value for subquery select statement as k-v tuple - synchronous."))
 
 ;; This exists because many different types of data structures in :select
 ;; clauses get implicit-grouping? called on them. So this defaults them to false.
 (extend-type #?(:clj Object :cljs object) ; https://cljs.github.io/api/cljs.core/extend-type
   ValueSelector
   (implicit-grouping? [_] false)
-  (format-value [_ _ _ _ _ _ _ _] nil))
+  (format-value [_ _ _ _ _ _ _ _] nil)
+  (solution-value [_ _ _] nil))
```

Comment:
If `solution-value` is in a new protocol, I don't think we should have a default implementation because there are some cases where it explicitly should not be allowed.

Reply:
Note previous related comment.

Reply:
Addressed in e041dc93c2e681f21fbbc5814cd98ae80aa956c9